drivers/hwinfo/hwinfo_mcux_syscon.c

```diff
@@ -22,7 +28,11 @@ ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 		length = DT_INST_REG_SIZE(0);
 	}
 
-	memcpy(buffer, uid_addr, length);
+	for(size_t i = 0 ; i < length / 4 ; i++) {
```

Comment:
Missed that. This could copy too less. I think it is faster to just copy all of them.

Reply:
Correct -- I wrote it to only copy the words that will `memcpy`'d into `buffer`. But I agree with you it is clearer to copy all the UUID words to the struct, and from there `memcpy` what will fit into `buffer`.

I have swapped the order of a `length` check and use the `dev_id.id` size, which is similar to other implementations and isolates the `DT_INST_REG_SIZE(0)` to a single instance.